### PR TITLE
[posix] clean up for radio url

### PR DESCRIPTION
**Pull request by bukepo**
* fix max-power-table documentation
* rename `baudrate` to `uart-baudrate`, `arg` to `forkpty-arg`
* update command line enumeration
* update README
* remove trailing argument from node.py

### CODE REVIEW

File: include/openthread/instance.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ extern "C" {
  * @note This number versions both OpenThread platform and user APIs.
  *
  */
-#define OPENTHREAD_API_VERSION (1)
+#define OPENTHREAD_API_VERSION (2)
```
Comment (jwhui) [inline include/openthread/instance.h:56]: Why does this PR or #4935 require a version bump? APIs have historically been constrained to [`include/`](https://github.com/openthread/openthread/tree/master/include). We need to have a clear description of the API surface and all such APIs documented in https://openthread.io/reference

Comment (bukepo) [reply include/openthread/instance.h:56]: Actually it's not for this PR, but the #4935 added posix platform API otSysGetRadioUrlHelpString(). I'm not sure if this should cause increment of this version number.

**Comment by wgtdkp** [comment]
@bukepo I got an exit code of value 5 (`OT_EXIT_ERROR_ERRNO`) by following this README with simulated RCP:
```shell
./bootstrap
$ make -f src/posix/Makefile-posix
$ make -f examples/Makefile-simulation
$ ./output/posix/x86_64-unknown-linux-gnu/bin/ot-cli 'spinel+hdlc+forkpty://output/x86_64-unknown-linux-gnu/bin/ot-rcp?arg=1'
$ echo $?
5
```

Any idea?

**Comment by wgtdkp** [comment]
> @bukepo I got an exit code of value 5 (`OT_EXIT_ERROR_ERRNO`) by following this README with simulated RCP:
> 
> ```shell
> ./bootstrap
> $ make -f src/posix/Makefile-posix
> $ make -f examples/Makefile-simulation
> $ ./output/posix/x86_64-unknown-linux-gnu/bin/ot-cli 'spinel+hdlc+forkpty://output/x86_64-unknown-linux-gnu/bin/ot-rcp?arg=1'
> $ echo $?
> 5
> ```
> 
> Any idea?

There is another wpantund process using node ID 1, fixed by stopping wpantund.

File: include/openthread/instance.h (hunk 1 above)
Comment (bukepo) [reply include/openthread/instance.h]: I reverted increasing the version number.

**Pull request merged**